### PR TITLE
fix: upgrade go to the latest go1.18

### PR DESCRIPTION
**Pull request by joker00777**
https://github.com/jenkins-x/jx/issues/8283

### CODE REVIEW

**Comment by joker00777** [comment]
@ankitm123 should we first start doing jx-logging jx-helpers , all the jx projects required in this and after that we should do this or is it cool ?

**Comment by ankitm123** [comment]
> should we first start doing jx-logging jx-helpers , all the jx projects required in this and after that we should do this or is it cool ?

Yes, we should start with the core packages.

**Comment by ankitm123** [comment]
Updated the docs: https://jenkins-x.io/community/maintainer_guide/go-upgrades/
Also post in the main issue, which package u are upgrading, so it's easier for others to keep track of the work.

**Review by rajatgupta24 — COMMENTED** [review]
Hello @joker00777 
This looks good to me and now core packages are also updated, so can you please make sure this is still good to go.
Thanks.

**Comment by joker00777** [comment]
Done @rajatgupta24 : ).

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ ORG := jenkins-x-plugins
 ORG_REPO := $(ORG)/$(NAME)
 RELEASE_ORG_REPO := $(ORG_REPO)
 ROOT_PACKAGE := github.com/$(ORG_REPO)
-GO_VERSION := 1.17.9
+GO_VERSION := 1.18.4
```
Comment (ankitm123) [inline Makefile:15]: Replace GO_VERSION with this section: https://github.com/jenkins-x-plugins/jx-pipeline/blob/89593c6e1e9d2c01b8a26296f53b6fdedcc48410/Makefile#L16-L22

File: .github/workflows/plugins-pr.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.17.9'
+          go-version: '1.18.4'
```
Comment (ankitm123) [inline .github/workflows/plugins-pr.yaml:22]: Update to 1.18.6

**Comment by ankitm123** [comment]
Change commit message (and PR title) to follow semantic convention

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**